### ISSUE TITLE
Buttons without discernible Text

### ISSUE THREAD

**Issue by krios82** [DFE-Digital/npd-find-and-explore#595]
# Automated Testing General Findings

## Buttons without discernible Text

Screen reader users are not able to discern the purpose of the button that has no accessible name. Therefore when the user focuses on this button the screen reader will simply read “button” (see image 1.1 – the grey box reading “button” is what the screen reader reads aloud). Unfortunately, there is not enough information for a user to understand the purpose and expected outcome of this search button. This does not meet the criteria set out in the WCAG 2.1 guidelines and therefore is a failure in this audit.

Success Criterion 4.1.2 Name, Role, Value (Level A): For all user interface components (including but not limited to: form elements, links and components generated by scripts);
* the name and role can be programmatically determined; 
* states, properties, and values that can be set by the user can be programmatically set; and 
* notification of changes to these items is available to user agents, including assistive technologies.

*Page Link:* https://find-npd-data.education.gov.uk/

![20200821 Accessibility Assurance Report - National Pupil Data docx](https://user-images.githubusercontent.com/2742327/92098346-0705f480-edd1-11ea-8900-f9594507607c.png)

## Required action

Add a discernible text to that button and inspect the whole application to verify if there's any other button in the application that needs to be fixed.

**Comment by dgamblin3008** [comment]
@cacquani after running through this with you yesterday on staging, I'm happy that the work's been done and I've tested it.

**Comment by dgamblin3008** [comment]
Awaiting the result of the accessibility test on 14/09/2020

**Comment by dgamblin3008** [comment]
The accessibility re-test picked up no issues, so this change is ready to push to production.

**Comment by dgamblin3008** [comment]
Deployed to production and tested successfully. Awaiting approval by PO, Gwen McGill, before closing the ticket.

**Comment by dgamblin3008** [comment]
PO has agreed to close the ticket.
